packages/manager/src/features/Volumes/Drawers/ManageTagsDrawer.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+
+import { ActionsPanel } from 'src/components/ActionsPanel/ActionsPanel';
+import { Drawer } from 'src/components/Drawer';
+import { Notice } from 'src/components/Notice/Notice';
+import { TagsInput } from 'src/components/TagsInput/TagsInput';
+import { useGrants } from 'src/queries/profile/profile';
+import { useUpdateVolumeMutation } from 'src/queries/volumes/volumes';
+
+import type { APIError, Volume } from '@linode/api-v4';
+
+interface Props {
+  onClose: () => void;
+  open: boolean;
+  volume: Volume | undefined;
+}
+
+export const ManageTagsDrawer = (props: Props) => {
+  const { onClose: _onClose, open, volume } = props;
+
+  const [generalError, setGeneralError] = useState<string>();
+
+  const { data: grants } = useGrants();
+
+  const { mutateAsync: updateVolume } = useUpdateVolumeMutation();
+
+  const isReadOnly =
+    grants !== undefined &&
+    grants.volume.find((grant) => grant.id === volume?.id)?.permissions ===
+      'read_only';
+
+  const {
+    control,
+    formState: { isDirty, isSubmitting },
+    handleSubmit,
+    reset,
+    setError,
+  } = useForm<{ tags: string[] }>({
+    values: { tags: volume?.tags ?? [] },
+  });
+
+  const onSubmit = handleSubmit(async (values) => {
+    try {
+      await updateVolume({
+        label: volume?.label ?? '',
+        tags: values.tags,
+        volumeId: volume?.id ?? -1,
+      });
+
+      onClose();
+    } catch (errors) {
+      errors.forEach((error: APIError) => {
+        if (error.field == 'tags') {
+          setError('tags', {
+            message: error.reason,
+          });
+        } else {
+          setGeneralError(
+            'Unable to edit this Volume at this time. Please try again later.'
+          );
```

Comment:
Rather than having a "generalError" state, we could use `react-hook-form`'s `root` error state. This is how we handle general errors on the Linode Create flow. 

We can also just use `error.reason` instead of that hardcoded error

```suggestion
          form.setError('root', { message: error.reason });
```

Reply:
@bnussman-akamai Hello, thank you for the hint, "root" has been used instead of the generalError.